### ISSUE TITLE
@babel/parser(ts): literal computed property names in enums

### ISSUE THREAD

**Issue by armano2** [babel/babel#12683]
## Bug Report

- [ ] I would like to work on a fix!

**Current behavior**
Babel fails if computed property is provided as key of enum, this synatx should be allowed only for computed properties with string literal value or template literal without arguments.

**Input Code**

- [REPL Babel](https://babeljs.io/repl#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&spec=false&loose=false&code_lz=KYDwDg9gTgLgBMAdgVwLZwGIQnA3gKDiLgG0ByAIwEMAvMgXQBp8BffIA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Cstage-2%2Ctypescript&prettier=false&targets=&version=7.12.12&externalPlugins=), [REPL Typescript](https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBMAdgVwLZwGIQnA3gWACg4S4BtAcgCMBDALwoF0AaIgXyKA)

```js
// should work
export enum Foo { ['baz'] }
export enum Foo { [`baz`] }

// should fail
export enum Foo { [foo] }               // Computed property names are not allowed in enums
export enum Foo { [1] }                 // An enum member cannot have a numeric name.
export enum Foo { [`test${foo}`] }      // Computed property names are not allowed in enums.
```

**Expected behavior**
Babel should allow computed property or template literals

**Note**
In my opinion this syntax is a issue in typescript as `Computed property names are not allowed in enums (1164)`

----
- Babel version(s): v7.12.0
- Node/npm version: 10-14
- OS: ----
- Monorepo: ----
- How you are using Babel: @babel/parser -> parse

**Comment by nicolo-ribaudo** [comment]
Do you think we should always allow them, or throw when `errorRecovery` is not `true` for the "should fail" cases you mentioned?

Imo this is more about syntax than about types because `enum Foo { [foo] }` is an error even if `type Foo = "foo"`.

**Comment by armano2** [comment]
> Imo this is more about syntax than about types because enum Foo { [foo] } is an error even if type Foo = "foo".

that's correct, this is just about a syntax, I'm unsure if babel actually wants to support this, as it looks like a issue on typescript side.

**Comment by armano2** [comment]
> Do you think we should always allow them, or throw when errorRecovery is not true for the "should fail" cases you mentioned?

with error recovery babel should let it parse as typescript does, this is useful scenario for consumers like prettier

**Comment by fedeci** [comment]
Should I submit a PR for this?

**Comment by armano2** [comment]
> Should I submit a PR for this?

yes :>
